habitat/tasks/nav/nav_task.py

```diff
@@ -363,6 +443,148 @@ def update_metric(self, episode, action):
             self._metric += 1
 
 
+class TopDownMap(habitat.Measure):
+    """Top Down Map measure
+    """
+
+    def __init__(self, sim: Simulator, config: Config):
+        self._sim = sim
+        self._draw_source_and_target = config.DRAW_SOURCE_AND_TARGET
+        self._draw_border = config.DRAW_BORDER
+        self._grid_delta = 3
+        self._step_count = 0
+        self._max_steps = config.MAX_EPISODE_STEPS
+        self._map_resolution = (config.MAP_RESOLUTION, config.MAP_RESOLUTION)
+        self._num_samples = 20000
+        self._ind_x_min = None
+        self._ind_x_max = None
+        self._ind_y_min = None
+        self._ind_y_max = None
+        self._previous_xy_location = None
+        self._coordinate_min = maps.COORDINATE_MIN
+        self._coordinate_max = maps.COORDINATE_MAX
+        self._top_down_map = None
+        self._cell_scale = (
+            self._coordinate_max - self._coordinate_min
+        ) / self._map_resolution[0]
+        super().__init__()
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "top_down_map"
+
+    def _check_valid_nav_point(self, point: List[float]):
+        return (
+            0.01
+            < self._sim.geodesic_distance(
+                point, [point[0], point[1] + 0.1, point[2]]
+            )
+            < 0.11
+        )
+
+    def get_original_map(self, episode):
+        top_down_map = maps.get_topdown_map(
+            self._sim,
+            self._map_resolution,
+            self._num_samples,
+            self._draw_border,
+        )
+
+        range_x = np.where(np.any(top_down_map, axis=1))[0]
+        range_y = np.where(np.any(top_down_map, axis=0))[0]
+
+        self._ind_x_min = range_x[0]
+        self._ind_x_max = range_x[-1]
+        self._ind_y_min = range_y[0]
+        self._ind_y_max = range_y[-1]
+
+        if self._draw_source_and_target:
+            # mark source point
+            s_x, s_y = maps.to_grid(
+                episode.start_position[0],
+                episode.start_position[2],
+                self._coordinate_min,
+                self._coordinate_max,
+                self._map_resolution,
+            )
+            point_padding = 2 * int(np.ceil(self._map_resolution[0] / 1250))
```

Comment:
Significane of 1250 here? Maybe you can use the value of resolution directly.

Reply:
1250 makes it invariant to the map_resolution changing in terms of the final image. It was just a value that looked pretty good. I can parameterize it.